atomic_reactor/plugins/post_tag_and_push.py

```diff
@@ -156,59 +149,43 @@ def get_repositories(self):
             "floating": floating_repositories,
         }
 
-    def run(self):
+    def run(self) -> Dict[str, Union[List, Dict[str, List[str]]]]:
+        is_source_build = PLUGIN_FETCH_SOURCES_KEY in self.workflow.data.prebuild_results
+
+        if not is_source_build and not is_flatpak_build(self.workflow):
+            self.log.info('not a flatpak or source build, skipping plugin')
+            return {'pushed_images': [],
+                    'repositories': self.get_repositories()}
+
         pushed_images = []
         wf_data = self.workflow.data
 
-        is_source_build = PLUGIN_FETCH_SOURCES_KEY in self.workflow.data.prebuild_results
-        if is_source_build:
-            source_docker_archive = self.workflow.build_dir.any_platform.exported_squashed_image
-            source_unique_image = self.source_get_unique_image()
-
         tag_conf = wf_data.tag_conf
-        if not tag_conf.unique_images:
-            if is_source_build:
-                tag_conf.add_unique_image(source_unique_image)
-            else:
-                tag_conf.add_unique_image(self.workflow.image)
 
-        image_size_limit = self.workflow.conf.image_size_limit
+        images = []
+        if is_source_build:
+            source_image = self.source_get_unique_image()
+            plugin_results = wf_data.buildstep_result[PLUGIN_SOURCE_CONTAINER_KEY]
+            image = plugin_results['image_metadata']
+            tag_conf.add_unique_image(source_image)
+            images.append((image, source_image))
+        else:
+            for image_platform in get_platforms(self.workflow.data):
+                plugin_results = wf_data.prepub_results[PLUGIN_FLATPAK_CREATE_OCI]
```

Comment:
flatpak_create_oci is supposed to be postbuild (related to https://github.com/containerbuildsystem/atomic-reactor/pull/1868)

(we should git rid of all the different plugin types, filed an internal story for that)

Reply:
fixed it